pkg/metaserver/spd/fetcher.go

```diff
@@ -159,6 +161,28 @@ func (s *spdFetcher) getSPDByNamespaceName(_ context.Context, namespace, name st
 	currentSPD := s.spdCache.GetSPD(key, true)
 	if currentSPD != nil {
 		return currentSPD, nil
+	} else if s.getFromRemote {
+		klog.Infof("[spd-manager] need to get spd %s from remote", key)
```

Comment:
this logic in the getSPDByNamespaceName can be abstracted into a function with the logic in sync.